lib/services/platform-service.ts

```diff
@@ -177,13 +177,62 @@ export class PlatformService implements IPlatformService {
 			platform = platform.toLowerCase();
 
 			this.preparePlatform(platform).wait();
-			if(options.emulator) {
+            if (options.emulator) {
 				this.deployOnEmulator(platform).wait();
 			} else {
 				this.deployOnDevice(platform).wait();
 			}
 		}).future<void>()();
-	}
+    }
+
+    public debugPlatform(platform: string): IFuture<void> {
+        return (() => {
+            platform = platform.toLowerCase();
+
+            if (options.emulator) {
+                this.debugOnEmulator(platform).wait();
+            } else {
+                this.debugOnDevice(platform).wait();
+            }
+        }).future<void>()();
+    }
+
+    public debugOnEmulator(platform: string): IFuture<void> {
+        return (() => {
+
+            // a bit redundant
+            this.deployOnEmulator(platform).wait();
```

Comment:
The code needs some refactoring. I tried to reuse logic to deployOn* related code path. In this case deployOnEmulator function. I guess I can split in better granularity and use the bits I need. Currently some of the logic will be performed twice.